jbmc/src/java_bytecode/java_bytecode_parser.cpp

```diff
@@ -1256,10 +1256,23 @@ void java_bytecode_parsert::rmethod_attribute(methodt &method)
           id2string(method.name)+":"+method.descriptor);
     }
 
-    // line number of method
+    // line number of method (the first line number available)
     if(!method.instructions.empty())
-      method.source_location.set_line(
-        method.instructions.begin()->source_location.get_line());
+    {
+      const auto it = std::find_if(
+        method.instructions.begin(),
+        method.instructions.end(),
+        [&](const instructiont &instruction) {
+          return !instruction.source_location.get_line().empty();
+        });
+      if(it != method.instructions.end())
+        method.source_location.set_line(it->source_location.get_line());
+    }
+
+    // method name
+    method.source_location.set_function(
```

Comment:
Might be worth factoring out into a separate function as this code is duplicated here: https://github.com/diffblue/cbmc/blob/develop/jbmc/src/java_bytecode/java_bytecode_parser.cpp#L1254

Reply:
Done, I moved the assignment to he method location higher up, and then just re-use it's value when assigning the instruction location.